packages/x-components/src/x-modules/url/store/getters/url-params.getter.ts

```diff
@@ -1,26 +1,30 @@
+import { MappedParams, UrlParamValue, UrlXStoreModule } from '../types';
 import { reduce } from '../../../../utils/object';
-import { UrlGetters, UrlParamValue, UrlXStoreModule } from '../types';
 
 /**
  * Default implementation for the {@link UrlGetters.urlParams} getter.
  *
  * @param state - Current {@link https://vuex.vuejs.org/guide/state.html | state} of the url module.
+ *
  * @returns The url params.
  *
  * @public
  */
 export const urlParams: UrlXStoreModule['getters']['urlParams'] = ({
-  config,
+  config: { urlParamNames },
   extraParams,
-  ...rawUrlParams
+  ...params
 }) => {
   return reduce(
-    { ...rawUrlParams, ...extraParams },
-    (urlParams, key, value) => {
-      const newKey = config.urlParamNames[key] ?? key;
-      urlParams[newKey] = value as UrlParamValue;
-      return urlParams;
+    { ...extraParams, ...params },
+    (mappedParams, key, value) => {
+      const mappedKey = urlParamNames[key] ?? key;
+      mappedParams[key] = {
+        key: mappedKey,
```

Comment:
I'm not 100% sure about this new abstraction... I think I now have the context to understand it, but I think it kinda divides the state handling into two different parts... On one hand we have the getter generating the URL params, but the action also decides what should go to a URL param and what is not.

For me the getter should generate the URL full dictionary of URL params, and the action should only take care of translating this object to the URL, without having different paths for different parameters. Single responsabilities for both of them. Then you can use an emitter to always keep the URL synced with this object generated by the getter. What do you think?

Reply:
I agree. This getter should return the final parameters to set un URL, removing for example the page when it is "1".

But I would also move the name mapping responsibility to the Action. This getter should return only an object with the parameters to set in URL, with the internal naming.

Then the action gets the configuration of mapping and updates the URL with the right naming for each parameter.